VMware.CloudFoundation.Reporting.psm1

```diff
@@ -536,64 +540,27 @@ Function Invoke-VcfConfigReport {
-                } else {
-                    $clusterDrsRuleHtml = Publish-ClusterDrsRule -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass -workloadDomain $workloadDomain
-                }
+                $clusterDrsRuleHtml = Invoke-Expression "Publish-ClusterDrsRule -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass $($commandSwitch)"; $reportData += $clusterDrsRuleHtml
 
                 # Collecting Resource Pool Details  Using PowerShell Functions
                 Write-LogMessage -Type INFO -Message "Generating the Resouce Pool Configuration for $workflowMessage."
-                if ($PsBoundParameters.ContainsKey("allDomains")) {
-                    $resourcePoolsHtml = Publish-ResourcePool -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass -allDomains
-                } else {
-                    $resourcePoolsHtml = Publish-ResourcePool -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass -workloadDomain $workloadDomain
-                }
+                $resourcePoolsHtml = Invoke-Expression "Publish-ResourcePool -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass $($commandSwitch)"; $reportData += $resourcePoolsHtml
 
                 # Collecting VM Overrides Using PowerShell Functions
                 Write-LogMessage -Type INFO -Message "Generating the VM Override Configuration for $workflowMessage."
-                if ($PsBoundParameters.ContainsKey("allDomains")) {
-                    $vmOverridesHtml = Publish-VmOverride -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass -allDomains
-                } else {
-                    $vmOverridesHtml = Publish-VmOverride -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass -workloadDomain $workloadDomain
-                }
+                $vmOverridesHtml = Invoke-Expression "Publish-VmOverride -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass $($commandSwitch)"; $reportData += $vmOverridesHtml
 
                 # Collecting Virtual Networking Using PowerShell Functions
                 Write-LogMessage -Type INFO -Message "Generating the Virtual Networking Configuration for $workflowMessage."
-                if ($PsBoundParameters.ContainsKey("allDomains")) {
-                    $vritualNetworkHtml = Publish-VirtualNetwork -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass -allDomains
-                } else {
-                    $vritualNetworkHtml = Publish-VirtualNetwork -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass -workloadDomain $workloadDomain
-                }
+                $vritualNetworkHtml = Invoke-Expression "Publish-VirtualNetwork -server $sddcManagerFqdn -user $sddcManagerUser -pass $sddcManagerPass $($commandSwitch)"; $reportData += $vritualNetworkHtml
```

Comment:
Minor, but we should search/replace `vritualNetworkHtml` with `virtualNetworkHtml`.

Reply:
But it doesn’t have impact as it’s consistent.

Reply:
Easily fixed only two places on this line after the refactor